### ISSUE TITLE
BayesianNetwork.from_samples error - TypeError: unorderable types: NoneType() < tuple()

### ISSUE THREAD

**Issue by tuxdna** [jmschrei/pomegranate#325]
While going through this [tutorial](https://github.com/jmschrei/pomegranate/blob/master/tutorials/Tutorial_4b_Bayesian_Network_Structure_Learning.ipynb), I encountered this error when using pomegranate on **Python 3.5**. 

```
In [27]: from sklearn.datasets import load_digits

In [28]: X, y = load_digits(10, True)

In [29]: X = X > numpy.mean(X)

In [30]: X = X[:,:18]

In [31]: model = BayesianNetwork.from_samples(X, algorithm='exact')
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-31-1fe501aa05d8> in <module>()
----> 1 model = BayesianNetwork.from_samples(X, algorithm='exact')

pomegranate/BayesianNetwork.pyx in pomegranate.BayesianNetwork.BayesianNetwork.from_samples()

pomegranate/BayesianNetwork.pyx in pomegranate.BayesianNetwork.discrete_exact_a_star()

pomegranate/utils.pyx in pomegranate.utils.PriorityQueue.pop()

TypeError: unorderable types: NoneType() < tuple()

```

This works fine on Python 2.7 .

**Comment by handsaway** [comment]
I c.an confirm I'm getting the same error on this tutorial on python 3.6 with pomegranate 0.7.7.

**Comment by Wesserg** [comment]
Yeah - same here. Works for exact-dp, greedy and chow-liu, but same error for "exact"

**Comment by willsilvano** [comment]
Same here :(

**Comment by poplarShift** [comment]
"NoneType() < tuple()" seems to be a Python 2 vs 3 compatibility issue. I've also noted all the tutorials being written with Python 2 in mind, IIRC. Are there other issues that require fixing for Python 3?

**Comment by jmschrei** [comment]
This should be fixed in v0.10.0. Please re-open if not.
